src/features/users/components/user-list-item.tsx

```diff
@@ -0,0 +1,220 @@
+import { Box, Divider, Grid, MenuItem, Paper, TextField, Typography } from "@mui/material";
+import { User } from "../../../app/models/User"
+import IconButton from '@mui/material/IconButton';
+import DeleteForeverIcon from '@mui/icons-material/DeleteForever';
+import { useState } from "react";
+import { userIdNameEmailHeaders, statusGenderHeaders } from "./user-headers";
+import FormikInputField from "../../../common-components/formik-field";
+import { useFormik } from "formik";
+import { GENDERS, STATUSES } from "../../../constants";
+import { useDispatch } from "react-redux";
+import { deleteUser, updateUser } from "../users-thunk";
+import { validationSchema } from "../../../app/commonSchemas.ts/schemas";
+
+// icons
+import EditIcon from '@mui/icons-material/Edit';
+import CloseIcon from '@mui/icons-material/Close';
+import CheckIcon from '@mui/icons-material/Check';
+
+interface userListItemProps {
+  user: User;
+  ShouldBeShaded: boolean;
+}
+
+export const UserListItem = (props: userListItemProps) => {
+  const [isEdit, setIsEdit] = useState(false);
+  const [isSaveButtonEnabled, setIsSaveButtonEnabled] = useState(false);
+
+  const dispatch = useDispatch();
+  const { user, ShouldBeShaded } = props;
+
+  const initialValuesEdit = {
+    email: '',
+    name: '',
+    status: '',
+    gender: '',
+  };
+
+  const checkIfIsValid = (value: any) =>
+    validationSchema.validate(value)
+    .then(() => setIsSaveButtonEnabled(true))
+    .catch(() => setIsSaveButtonEnabled(false));
+
+  const formik = useFormik({
+    initialValues: initialValuesEdit,
+    validationSchema,
+    validate: checkIfIsValid,
+    onSubmit: () => {},
+  });
+
+  const handleSaveEditUser = async () => {
+    const postBody = {
+      name: formik.values?.name,
+      email: formik.values?.email,
+      gender: formik.values?.gender,
+      status: formik.values?.status,
+    } as any
+
+    dispatch(updateUser({userId: user.id, postBody }))
+    setIsEdit(false);
+  };
+
+  const usersList = (
+    <div style={{margin: '50px'}}>
+      {user?.id && (
+        <Box padding={2} paddingBottom={4}>
+          <div>
+            <Grid container spacing={1}>
+              {userIdNameEmailHeaders}
+              <Grid item xs={4}>
+                <Typography>
+                  {user.id}
+                </Typography>
+              </Grid>
+              <Grid item xs={4}>
+                <Typography>
+                  {user.email}
+                </Typography>
+              </Grid>
+              <Grid item xs={4}>
+                <Typography>
+                  {user.name}
+                </Typography>
+              </Grid>
+              <Grid item xs={12}>
+                <Divider />
+              </Grid>
+              {statusGenderHeaders}
+              <Grid item xs={6}>
```

Comment:
Adjust the grid size on the Gender and Status to handle responsiveness accordingly

example:
<Grid item xs={12} md={6}>